src/raftstore/store/store.rs

```diff
@@ -1772,6 +1773,71 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         };
     }
 
+    fn on_compaction_finished(&mut self, event: CompactedEvent) {
+        // If size declining is trivial, skip.
+        let total_bytes_declined = if event.total_input_bytes > event.total_output_bytes {
+            event.total_input_bytes - event.total_output_bytes
+        } else {
+            0
+        };
+        if total_bytes_declined < self.cfg.region_split_check_diff.0
+            || total_bytes_declined * 10 < event.total_input_bytes
+        {
+            return;
+        }
+
+        let output_level_str = event.output_level.to_string();
+        COMPACTION_DECLINED_BYTES
+            .with_label_values(&[&output_level_str])
+            .observe(total_bytes_declined as f64);
+
+        // Calculate influenced regions.
+        let mut influenced_regions = vec![];
+        if let Some((end_key, region_id)) = self.region_ranges
+            .range((Included(event.end_key.clone()), Unbounded))
```

Comment:
Because `event.end_key` is inclusive?

Reply:
Because we only store end_key in `self.region_ranges`. For example we have two regions 1 [0, 50) and 2 [50, 100), so (50, 1) and (100, 2) will be stored in `self.region_ranges`. If we have a compaction whose smallest key is 20 and largest key is 60, `self.region_ranges.range((Included(20), Included(60)))` only return (50, 1).